### DISCUSSION TITLE
How can i fire actions(cancel,finish,remove Last vertex) programatically  ?

### DISCUSSION THREAD

**Discussion by anujs099** [geoman-io/leaflet-geoman#1469] [Questions & Help]
hey guys !  need help, i want to fire finish , remove last vertex and cancel actions programatically 
@Falke-Design

**Comment by Falke-Design** [comment]
Currently only `cancel` is supported offical with `map.pm.disableDraw()`.

To remove the last vertex / finish the shape you can use the internal function, but the function can be changed in future.

```
map.pm.Draw.Line._removeLastVertex();
map.pm.Draw.Line._finishShape();
```
or
```
map.pm.Draw.Polygon._removeLastVertex();
map.pm.Draw.Polygon._finishShape();
```

Please also look into #861
